src/test/util/setup_common.cpp

```diff
@@ -355,6 +355,43 @@ CMutableTransaction TestChain100Setup::CreateValidMempoolTransaction(CTransactio
     return mempool_txn;
 }
 
+std::vector<CTransactionRef> TestChain100Setup::PopulateMempool(FastRandomContext& det_rand, size_t num_transactions, bool submit)
+{
+    std::vector<CTransactionRef> mempool_transactions;
+    std::deque<std::pair<COutPoint, CAmount>> unspent_prevouts;
+    std::transform(m_coinbase_txns.begin(), m_coinbase_txns.end(), std::back_inserter(unspent_prevouts),
+        [](const auto& tx){ return std::make_pair(COutPoint(tx->GetHash(), 0), 50 * COIN); });
+    CScript witnessScript = CScript() << OP_DROP << OP_TRUE;
+    CScript scriptPubKey = GetScriptForDestination(WitnessV0ScriptHash(witnessScript));
+    while (num_transactions > 0 && !unspent_prevouts.empty()) {
+        CMutableTransaction mtx = CMutableTransaction();
+        const auto& [prevout, amount] = unspent_prevouts.front();
+        mtx.vin.push_back(CTxIn(prevout, CScript()));
+        // Create a 1-input, n-output transaction.
+        // The number of outputs is random, between 1 and 24.
+        const size_t num_outputs = det_rand.randrange(24) + 1;
+        // Approximately 1000sat "fee," equal output amounts.
+        const CAmount amount_per_output = (amount - 1000) / num_outputs;
+        for (size_t n{0}; n < num_outputs; ++n) {
+            CScript spk = CScript() << CScriptNum(num_transactions + n);
+            mtx.vout.push_back(CTxOut(amount_per_output, spk));
+        }
+        unspent_prevouts.pop_front();
+        CTransactionRef ptx = MakeTransactionRef(mtx);
+        mempool_transactions.push_back(ptx);
```

Comment:
would it be better to do a swap with random index after this push_back?

That way you get a bit more of a 'arbitrary' txn graph, v.s. this one which will have a very specific structure.

Reply:
Oh, I think you mean random swap for `unspent_prevouts` after insert? Added that. I think we have a decent amount of randomness in the structure that way.

Reply:
correct. random swap with unspent_prevouts.